src/main/splitting/splitting.py

```diff
@@ -0,0 +1,106 @@
+    language = get_ct_language(data)
+    # add unique to logs
+    log.info(file_log_info + ", language is " + language + ", found " + str(data.shape[0]) + " fragments")
+
+    if language is not consts.LANGUAGE.NOT_DEFINED.value:
+        unique_fragments = list(data[FRAGMENT].unique())
+        log.info("Found " + str(len(unique_fragments)) + " unique fragments")
+
+        fragment_to_test_results_dict = dict(map(lambda f: (f, check_tasks(tasks, f, in_and_out_files_dict, language)), unique_fragments))
+        data[consts.CODE_TRACKER_COLUMN.TESTS_RESULTS.value] = data.apply(lambda row:
+                                                                          fragment_to_test_results_dict[row[FRAGMENT]], axis=1)
+    else:
+        data[consts.CODE_TRACKER_COLUMN.TESTS_RESULTS.value] = str([consts.TEST_RESULT.LANGUAGE_NOT_DEFINED.value] * len(tasks))
+
+    return language, data
+
+
+# since lists of tasks have small size, it should work faster than creating a set
+def intersect(list_1: list, list_2: list):
+    return [e for e in list_1 if e in list_2]
+
+
+# first version of splitting
+def find_real_splits(supposed_splits: list):
+    real_splits = []
+    if len(supposed_splits) == 0:
+        return real_splits
+
+    prev_split = supposed_splits[0]
+    prev_intersected_tasks = prev_split[consts.SPLIT_DICT.TASKS.value]
+
+    for curr_split in supposed_splits:
+        curr_intersected_tasks = intersect(prev_intersected_tasks, curr_split[consts.SPLIT_DICT.TASKS.value])
+        if len(curr_intersected_tasks) == 0:
+            # it means that the supposed task has changed, so we should split on prev_split
+            real_splits.append({consts.SPLIT_DICT.INDEX.value: prev_split[consts.SPLIT_DICT.INDEX.value],
+                                consts.SPLIT_DICT.RATE.value: prev_split[consts.SPLIT_DICT.RATE.value],
+                                consts.SPLIT_DICT.TASKS.value: prev_intersected_tasks})
+            prev_intersected_tasks = curr_split[consts.SPLIT_DICT.TASKS.value]
+        else:
+            prev_intersected_tasks = curr_intersected_tasks
+
+        prev_split = curr_split
+
+    # add the last split
+    real_splits.append({consts.SPLIT_DICT.INDEX.value: prev_split[consts.SPLIT_DICT.INDEX.value],
+                        consts.SPLIT_DICT.RATE.value: prev_split[consts.SPLIT_DICT.RATE.value],
+                        consts.SPLIT_DICT.TASKS.value: prev_intersected_tasks})
+    return real_splits
+
+
+def filter_already_tested_files(files: list, result_folder_path: str):
+    tested_files = get_all_file_system_items(result_folder_path, ct_file_condition, consts.FILE_SYSTEM_ITEM.FILE.value)
+    # to get something like 'ati_239/Main_2323434_343434.csv'
+    tested_folder_and_file_names = list(map(lambda f: get_parent_folder_name(f) + "/" + get_file_name_from_path(f), tested_files))
+    return list(filter(lambda f: get_parent_folder_name(f) + "/" + get_file_name_from_path(f) not in tested_folder_and_file_names, files))
```

Comment:
I think we can define an internal function, which calculates `get_parent_folder_name(f) + "/" + get_file_name_from_path(f)` and use it in the 79 and 80 rows